cmd/gateway/s3x/gateway-s3x-object.go

```diff
@@ -176,42 +144,41 @@ func (x *xObjects) PutObject(
 			obinfo.ContentType = v
 		}
 	}
+	return obinfo
+}
+
+// PutObject creates a new object with the incoming data,
+func (x *xObjects) PutObject(
+	ctx context.Context,
+	bucket, object string,
+	r *minio.PutObjReader,
+	opts minio.ObjectOptions,
+) (minio.ObjectInfo, error) {
+	err := x.ledgerStore.AssertBucketExits(bucket)
+	if err != nil {
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, "", "")
+	}
+
 	data, err := ioutil.ReadAll(r)
 	if err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, object, "")
 	}
-	obinfo.Size_ = int64(len(data))
-	dataHash, err := x.dagPut(ctx, data)
+	obinfo := newObjectInfo(bucket, object, len(data), opts)
+	dataHash, err := ipfsSaveBytes(ctx, x.dagClient, data)
 	if err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, object, "")
 	}
-	// add the object to ipfs
-	objectHash, err := x.objectToIPFS(ctx, &Object{
+	// add the object to bucket
+	err = x.ledgerStore.PutObject(ctx, bucket, object, &Object{
 		DataHash:   dataHash,
 		ObjectInfo: obinfo,
 	})
 	if err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
-	}
-	// update the bucket on ipfs with the new object
-	bucketHash, err := x.addObjectToBucketAndIPFS(ctx, object, objectHash, bucket)
-	if err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, object, "")
 	}
-	// update internal ledger state with bucket hash
-	if err := x.ledgerStore.UpdateBucketHash(bucket, bucketHash); err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
-	}
-	// update internal ledger state with the new object
-	if err := x.ledgerStore.AddObjectToBucket(bucket, object, objectHash); err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
-	}
-	log.Printf(
-		"bucket-name: %s, bucket-hash: %s, object-name: %s, object-hash: %s",
-		bucket, bucketHash, object, objectHash,
-	)
+	log.Printf("bucket-name: %s, object-name: %s", bucket, object)
```

Comment:
This should log the new bucket hash, and the object hash.

Reply:
I removed it because retrieving it was tedious using the new code, I can add it back if necessary.

Reply:
Yea I can see that. leave it out then.